### PR TITLE
Nameres: fix multidim array aggregates resolution

### PR DESCRIPTION
**Pull request by raph-amiard**

### CODE REVIEW

File: ada/language/ast.py [hunk 1]
```diff
@@ -6159,24 +6169,26 @@ class BaseAggregate(Expr):
...
-        return Self.direct_parent_aggregate.then(
-            lambda p: p.sub_aggregate_rank(r + 1),
-            default_val=r
+    @langkit_property(return_type=MultidimAggregateInfo, dynamic_vars=[origin])
+    def multidim_root_aggregate(r=(Int, 0)):
+        """
+        Return the root parent aggregate if Self is part of a multidimensional
+        array aggregate (either the root or a sub-aggregate).
+        """
+        return Entity.type_val.cast(BaseTypeDecl).then(
```
Comment (Roldak) [inline ada/language/ast.py]: Can you explain this logic? What kind of value can "type_val" take for nested aggregates?

File: ada/language/ast.py [hunk 2]
```diff
@@ -6159,24 +6169,26 @@ class BaseAggregate(Expr):
...
+            lambda tv: If(
+                tv.array_ndims > 1,
+                MultidimAggregateInfo.new(agg=Entity, typ=tv, rank=r),
+                # If we're here, we found a type, and it's not a multidim
+                # array: Stop there.
+                No(T.MultidimAggregateInfo)
+            ),
+            # If we're here, there is a parent aggregate and no type_val:
+            # recurse up.
+            default_val=Entity.Entity.parent.parent.parent.cast(T.Aggregate)
```
Comment (Roldak) [inline ada/language/ast.py]: Entity.Entity?

File: ada/language/ast.py (hunk 1 above)
Comment (raph-amiard) [reply ada/language/ast.py]: nested aggregates of a multidimensional array have no types, so type_val is empty

File: ada/language/ast.py (hunk 2 above)
Comment (raph-amiard) [reply ada/language/ast.py]: fixid

File: ada/language/ast.py (hunk 1 above)
Comment (raph-amiard) [reply ada/language/ast.py]: Added some comments !

**Pull request merged**